src/Engines/WebGPU/webgpuShaderProcessors.ts

```diff
@@ -366,7 +367,11 @@ export class WebGPUShaderProcessor implements IShaderProcessor {
         if (!isFragment) {
             const lastClosingCurly = code.lastIndexOf("}");
             code = code.substring(0, lastClosingCurly);
-            code += "gl_Position.y *= -1.;\ngl_Position.z = (gl_Position.z + gl_Position.w) / 2.0; }";
+            code += "gl_Position.y *= -1.;\n";
+            if (!engine.isNDCHalfZRange) {

```

Comment:
maybe passing the flag instead of the engine will make it a bit abstracter? Just a thought.

Reply:
The `engine` is already a parameter of `postProcessor` (see the [interface](https://github.com/BabylonJS/Babylon.js/blob/master/src/Engines/Processors/iShaderProcessor.ts#L15)), I simply didn't need it yet in the function.